### PR TITLE
Updating historical calibrated data

### PR DESCRIPTION
**Pull request by NoahNsimbe**
**_WHAT DOES THIS PR DO?_**
Updating BigQuery hourly measurements table to include calibrated data.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[PLAT-1099](https://airqoteam.atlassian.net/browse/PLAT-1099)

### CODE REVIEW

File: src/airflow/airflow_xcom/gcs_xcom_backend.py [hunk 1]
```diff
@@ -71,3 +71,10 @@ def deserialize_value(result) -> Any:
             result.reset_index(drop=True, inplace=True)
 
         return result
+
+    def orm_deserialize_value(self) -> Any:
+        value = GCSXComBackend.deserialize_value(self)
+        if isinstance(value, pd.DataFrame):
+            return value.to_dict(orient="records")
+
+        return super().orm_deserialize_value()
```
Comment (NoahNsimbe) [inline src/airflow/airflow_xcom/gcs_xcom_backend.py:80]: XCom ORM Serialization for Airflow

File: src/airflow/main.py [hunk 2]
```diff
@@ -386,6 +389,114 @@ def calibrate_historical_data(start_date_time, end_date_time, tenant):
...
+        3,
+    ):
+        print(merged_data.head(8))
+
+    big_query_api = BigQueryApi()
+    big_query_api.save_data(
+        dataframe=merged_data,
+        table=big_query_api.hourly_measurements_table,
+        job_action=JobAction.OVERWRITE,
+    )
```
Comment (NoahNsimbe) [inline src/airflow/main.py:497]: overwriting hourly measurements table with calibrated data

**Comment by NoahNsimbe** [comment]
Hi @Baalmart and @sserurich . Am getting this PR to production inorder to address the urgent data requests.

File: src/airflow/dags/data_warehouse.py [hunk 3]
```diff
@@ -78,11 +80,22 @@ def merge_data(
...
 
         big_query_api = BigQueryApi()
-        big_query_api.save_data(dataframe=data, table=big_query_api.analytics_table)
+        big_query_api.reload_data(
+            dataframe=data,
+            table=big_query_api.analytics_table,
+            start_date_time=start_date_time,
+            end_date_time=end_date_time,
+            tenant="airqo",
+        )
```
Comment (NoahNsimbe) [inline src/airflow/dags/data_warehouse.py:98]: reloading the datawarehouse table(analytics table) with calibrated values

File: src/airflow/dags/meta_data.py [hunk 4]
```diff
@@ -36,7 +36,7 @@ def load(data: pd.DataFrame):
         from airqo_etl_utils.constants import JobAction
 
         big_query_api = BigQueryApi()
-        big_query_api.save_data(
+        big_query_api.load_data(
             dataframe=data,
             table=big_query_api.sites_table,
             job_action=JobAction.OVERWRITE,
```
Comment (Baalmart) [inline src/airflow/dags/meta_data.py:42]: Thanks @NoahNsimbe  ...interested to continue this discussion about overwrite/update of historical data.
I am starting to get uncomfortable with it.
I sense a future where we do not have to update or overwrite existing time series data.......happy to hear suggestions on this.

**Review by Baalmart — APPROVED** [review]
LGTM, just left a comment for future review.

Thanks!

**Pull request merged**